File: clients/android/src/co/viewfinder/ContentTable.java
// Copyright 2013 Viewfinder. All rights reserved.
// Author: Peter Mattis
package co.viewfinder;

public class ContentTable {
  private final long mNativePointer;

  ContentTable(long nativePointer) {
    mNativePointer = nativePointer;
  }

  protected final byte[] loadContent(long contentId, DB db) {
    return LoadContent(mNativePointer, contentId, db.nativeHandle());
  }

  protected native byte[] LoadContent(long native_ptr, long content_id, long db_handle);
}

File: clients/shared/ContentTable.cc
// Copyright 2012 Viewfinder. All rights reserved.
// Author: Peter Mattis.

#import "ContentTable.h"
#import "StringUtils.h"

string EncodeContentKey(const string& prefix, int64_t local_id) {
  return prefix + ToString(local_id);
}

string EncodeContentServerKey(const string& prefix, const string& server_id) {
  return prefix + server_id;
}

// local variables:
// mode: c++
// end:
